applications/CoSimulationApplication/custom_data_structure/pyKratos/Variable.py

```diff
@@ -44,55 +46,32 @@ def __str__(self):
         return 'Variable-Component "{}"'.format(self.Name())
 
 
-# TODO maybe separate the class-declarations from the variable declarations?
-def CreateDoubleVariable(name):
-    if name in globals():
+def CreateDoubleVariable(module, name):
+    if name in KM._registered_variables:
         raise NameError('Variable "{}" exists already!'.format(name))
-    globals()[name] = Variable(name, "Double", 0.0)
 
-def CreateComponentVariable(name, source_variable, component_index):
-    if name in globals():
+    KM._registered_variables[name] = Variable(name, "Double", 0.0)
+    setattr(module, name, KM._registered_variables[name])
+
+def CreateComponentVariable(module, name, source_variable, component_index):
+    if name in KM._registered_variables:
         raise NameError('Variable "{}" exists already!'.format(name))
-    globals()[name] = VariableComponent(name, source_variable, component_index)
+    KM._registered_variables[name] = VariableComponent(name, source_variable, component_index)
+    setattr(module, name, KM._registered_variables[name])
 
-def CreateArray3Variable(name):
-    if name in globals():
+def CreateArray3Variable(module, name):
```

Comment:
What does this module do ?
you are not using it here right ?

Reply:
it is used, in line 68

Reply:
fyi this is necessary for having "application" variables => e.g. `KratosMultiphysics.TIME` or `KratosMultiphysics.CoSimulationApplication.MY_COSIM_VAR`